scripts/generator/graphql_generator/csharp/SDK/Android/AndroidWebCheckout.cs.erb

```diff
@@ -0,0 +1,41 @@
+namespace <%= namespace %>.SDK.Android {
+    #if UNITY_ANDROID
+    using System;
+    using <%= namespace %>.SDK;
+
+    class AndroidWebCheckout: WebCheckout {
+
+        private ShopifyClient _client;
+        private Cart _cart;
+
+        protected override ShopifyClient Client {
+            get {
+                return _client;
+            }
+        }
+
+        protected override Cart Cart {
+            get {
+                return _cart;
+            }
+        }
+
+        private const string UnityPlayerClassName = "com.unity3d.player.UnityPlayer";
+        private const string WebCheckoutSessionClassName = "com.shopify.unity.buy.WebCheckoutSession";
+
+        public AndroidWebCheckout(Cart cart, ShopifyClient client) {
+            _cart = cart;
+            _client = client;
+        }
+
+        public void Checkout(string checkoutURL, CheckoutSuccessCallback success, CheckoutCancelCallback cancelled, CheckoutFailureCallback failure) {
+            #if !SHOPIFY_MONO_UNIT_TEST
+            using(var webCheckoutSession = new AndroidJavaObject(WebCheckoutSessionClassName)) {
```

Comment:
`AndroidWebCheckout` will always be around as soon as `Cart` is instantiated, so we can't necessarily keep the reference till `AndroidWebCheckout` is gone. And since we don't provide any guarantees that only one `Checkout` is called at a time, I don't see the benefit of holding a reference to an unmanaged object. Are there any other concerns?